src/etc/is.ts

```diff
@@ -144,6 +144,10 @@ export function isUnaryExpression(node: TSESTree.Node): node is TSESTree.UnaryEx
   return node.type === AST_NODE_TYPES.UnaryExpression;
 }
 
+export function isUnionType(node?: TSESTree.Node): node is TSESTree.TSUnionType {
+  return node === undefined ? false : node.type === AST_NODE_TYPES.TSUnionType;
+}
```

Comment:
```suggestion
export function isUnionType(node: TSESTree.Node): node is TSESTree.TSUnionType {
  return node.type === AST_NODE_TYPES.TSUnionType;
}
```

Since every other function in this file does not check for `undefined`, can we just do `typeArgs && isUnionType(typeArgs)` in the rule itself?  This would help consistency.

Reply:
I wondered about this, yeah good idea. Ill take this suggestion

Reply:
(Typecheck failed because you'll need to null-check in the rule itself)

Reply:
Ha, yep - fixed now but the checks don't seem to be running now